src/vm/compiler.rs

```diff
@@ -120,8 +121,10 @@ impl Compiler {
                     // Get class `Value` by name and push it on to the VM's stack
                     self.get_variable(&class.name, span, gc)?;
 
-                    for (field_assign, span) in &class.fields {
-                        let name = &field_assign.identifier.name;
+                    for field_assign in &class.fields {
+                        let (assign, span) = &field_assign.field;
+
+                        let name = &assign.identifier.name;
```

Comment:
Updating for the new class field data structure which contains the previous field info + access modifier info